src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/dft_ref.cl

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// alternative: https://github.com/OpenCL/ComplexMath/blob/master/clcomplex.h
+typedef float2 cfloat;
+#define real(a) ((a).s0)
+#define imag(a) ((a).s1)
+#define cmult(a, b) ((cfloat)(real(a) * real(b) - imag(a) * imag(b), real(a) * imag(b) + imag(a) * real(b)))
+#define crmult(a, b) ((cfloat)(real(a) * (b), imag(a) * (b)))
+#define cadd(a, b) ((cfloat)(real(a) + real(b), imag(a) + imag(b)))
+#define expi(x) ((cfloat)(cos(x), sin(x)))
+#define expmi(x) ((cfloat)(cos(x), -sin(x)))
+#define cload(p, offset, pitch) ((cfloat)((p)[offset], (p)[(offset)+(pitch)]))
+#define cstore(p, offset, pitch, x) ((p)[offset] = real(x), (p)[(offset)+(pitch)] = imag(x))
+#define czero() ((cfloat)(0))
+
+// TODO: pregenerate e{r,i} array on host in macro. maybe it could be done with kernel which runs once?
+KERNEL (dft_ref)(const __global INPUT0_TYPE *data, __global OUTPUT_TYPE *output)
+{
+    size_t g2 = get_global_id(2);
+    ptrdiff_t y1 = g2 % OUTPUT_SIZES[1];
+#if OUTPUT_DIMS > 2
+    g2 /= OUTPUT_SIZES[1];
+    ptrdiff_t y2 = g2 % OUTPUT_SIZES[2];
+#if OUTPUT_DIMS > 3
+    g2 /= OUTPUT_SIZES[2];
+    ptrdiff_t y3 = g2 % OUTPUT_SIZES[3];
+#if OUTPUT_DIMS > 4
```

Comment:
its only supporting bfzyx, do we need to divide these cases?

Reply:
tests only test one rank, cl kernel supports fewer dimensions and can be easily extended to support more. if it's known for sure that we will never need other ranks, i can get rid of some ifdefs in kernel